src/main/groovy/edu/oregonstate/mist/trailsapi/db/TrailDAO.groovy

```diff
@@ -0,0 +1,42 @@
+package edu.oregonstate.mist.trailsapi.db
+
+import edu.oregonstate.mist.trailsapi.core.Trail
+import edu.oregonstate.mist.trailsapi.mapper.TrailMapper
+import org.skife.jdbi.v2.sqlobject.SqlUpdate
+import org.skife.jdbi.v2.sqlobject.customizers.RegisterMapper
+import org.skife.jdbi.v2.sqlobject.SqlQuery
+import org.skife.jdbi.v2.sqlobject.Bind
+import org.skife.jdbi.v2.sqlobject.BindBean
+
+@RegisterMapper(TrailMapper)
+interface TrailDAO extends Closeable {
+	@SqlUpdate ("""
+		INSERT INTO TRAILS (ID, NAME, ZIPCODE, DIFFICULTY_ID, LARGEDROP, SMALLDROP, WOODRIDE, SKINNY,
+		LARGEJUMP, SMALLJUMP, GAP)
+		VALUES (
+				(:id),
+				(:trail.name),
+				(:trail.zipCode),
+				(SELECT DIFFICULTY_ID FROM TRAIL_DIFFICULTIES WHERE DIFFICULTY_COLOR = :trail.difficulty),
+				(:trail.largeDrop),
+				(:trail.smallDrop),
+				(:trail.woodRide),
+				(:trail.skinny),
+				(:trail.largeJump),
+				(:trail.smallJump),
+				(:trail.gap)
+			)
+	""")
+	void postTrail(@Bind("id") Integer id, @BindBean("trail") Trail trail)
+
+	@SqlQuery("SELECT TRAIL_SEQ.NEXTVAL FROM DUAL")
+	Integer getNextId()
+
+	@SqlQuery("""SELECT ID FROM TRAILS WHERE
+				NAME = :trail.name
```

Comment:
You're comparing the exact values here, which is fine, but this means if a trail like `Whistlepunk` exists and I'm trying to create `whistlepunk`, (with all else being equal), this method won't consider it a duplicate. You might want to do a case-insensitive comparison, but I'm good either way. For example, you could do `UPPER(NAME) = UPPER(trail.name)`

Reply:
That is a good point. I have implemented the case-insensitive comparison in the "Improve getConflictingTrails" commit. I have removed single quotes and spaces from both strings as well, so that "Phil's Trail" conflicts with "Phils Trail", and "Whistlepunk" conflicts with "Whitstle Punk".